src/api/socket.c

```diff
@@ -312,10 +315,29 @@ static void *telnet_connection_handler_thread(void *socket_desc)
 	char client_message[SOCKETBUFFERLEN] = "";
 
 	// Set thread name
-	char threadname[16];
+	char threadname[16] = { 0 };
 	sprintf(threadname, "telnet-%i", sock);
 	prctl(PR_SET_NAME, threadname, 0, 0, 0);
-	//Receive from client
+
+	// Store TID of this thread
+	lock_shm();
+	unsigned int tid;
+	for(tid = 0; tid < MAX_API_THREADS; tid++)
+	{
+		if(api_threads[tid] == 0)
+		{
+			api_threads[tid] = pthread_self();
+			break;
+		}
+	}
+	unlock_shm();
+	if(tid == MAX_API_THREADS)
```

Comment:
Is this code supposed to be a duplicate (ish) of what Dan commented on below? (lines 412-416)

Reply:
Thanks, updated.